api/app/Models/Pool.php

```diff
@@ -150,4 +149,23 @@ public function scopeWasPublished(Builder $query, ?array $args)
         $query->where('published_at', '<=', Carbon::now()->toDateTimeString());
         return $query;
     }
+
+    public function scopeAvailableToQuery(Builder $query)
+    {
+        $userId = Auth::user()->id;
+        $user = User::find($userId);
+
+        if ($user->isAbleTo("view-team-pool")) {
+            $teamIds = $user->rolesTeams()->get()->pluck('id');
+            return $query->whereHas('team', function (Builder $query) use ($teamIds) {
+                return $query->whereIn('id', $teamIds);
+            });
+        }
```

Comment:
I know we only have one team role right now, buuuut... this might not generalize, since _technically_ you might not have the view-team-pool permission on of your teams. You'd have to check `isAbleTo("view-team-pool", $teamId)`, and only add it to teamIds if true.

Reply:
Okay, I hope this makes sense. There are two checks (in the interest of not looping through teams when unnecessary).

[make pool team scope more strict](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6064/commits/159c55cb87ebbab945b0756f067ad054cbf1b65c)